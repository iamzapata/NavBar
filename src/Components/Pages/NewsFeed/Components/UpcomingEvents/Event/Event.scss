@import "styles/variables.scss";

.Event {
  margin: 0 0 10px 0;
  .title {
    font-weight: normal;
    color: $color-black-text;
  }

  .subtitle {
    color: $color-gray-content-text;
  }
}
