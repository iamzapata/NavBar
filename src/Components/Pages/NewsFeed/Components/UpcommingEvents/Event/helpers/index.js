import { format, isSameWeek } from "date-fns"

export default (friendlyEventDate = date => {
  if (isSameWeek(date, new Date())) return format(date, "dddd, h A")

  return format(date, "MMMM D, h A")
})
