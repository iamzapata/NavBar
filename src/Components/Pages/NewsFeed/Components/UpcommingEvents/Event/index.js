import { format, isSameWeek } from "date-fns"

export default const friendlyEventDate = date => {
  if (isSameWeek(date, new Date())) return format(date, "dddd, h A")

  return format(date, "MMMM D, h A")
}
