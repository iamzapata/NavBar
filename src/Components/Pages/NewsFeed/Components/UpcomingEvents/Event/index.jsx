import React from "react"
import { shape, string } from "prop-types"
import friendlyEventDate from "./helpers"
import "./Event.scss"

const Event = ({ event: { name, date, location } }) => {
  return (
    <div className="Event">
      <p className="Event__Name title is-size-6">{name}</p>
      <p className="subtitle is-size-7">
        <span className="Event__Date">{friendlyEventDate(date)}</span>
        {", "}
        <span className="Event__Location">{location}</span>
      </p>
    </div>
  )
}

Event.propTypes = {
  event: shape({
    name: string.isRequired,
    date: string.isRequired,
    location: string.isRequired
  }).isRequired
}

export default Event
