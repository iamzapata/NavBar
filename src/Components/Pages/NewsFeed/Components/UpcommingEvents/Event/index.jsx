import React from "react"
import { shape, string } from "prop-types"
import { isSameWeek, format } from "date-fns"
import "./Event.scss"

const Event = ({ event: { name, date, location } }) => {
  return (
    <div className="Event">
      <p className="Event__Name title is-size-6">{name}</p>
      <p className="subtitle is-size-7">
        <span className="Event__Date">{date}</span>
        {", "}
        <span className="Event__Location">{location}</span>
      </p>
    </div>
  )
}

Event.propTypes = {
  event: shape({
    name: string,
    date: string,
    location: string
  })
}

export default Event
