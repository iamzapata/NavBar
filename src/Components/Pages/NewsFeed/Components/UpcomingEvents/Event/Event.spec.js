import React from "react"
import { render, cleanup } from "react-testing-library"
import Event from "./index"

describe("<Event />", () => {
  afterEach(cleanup)
  it("Should render self", () => {
    const { container } = render(<Event />)
    expect(container.firstChild).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })
})
