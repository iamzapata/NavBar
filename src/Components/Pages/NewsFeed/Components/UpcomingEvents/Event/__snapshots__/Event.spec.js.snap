// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Event /> Should render self 1`] = `
<div
  class="Event"
>
  <p
    class="Event__Name title is-size-6"
  />
  <p
    class="subtitle is-size-7"
  >
    <span
      class="Event__Date"
    >
      Invalid Date
    </span>
    , 
    <span
      class="Event__Location"
    />
  </p>
</div>
`;
